projects/bash_networking_security/tlsHandshake.sh

```diff
@@ -1 +1,54 @@
-#!/bin/bash
+#!/bin/bash -x
+
+
+# Step 1 - Client Hello (Client -> Server)
+RESPONSE=$(curl -X POST -H "Content-Type: application/json" -d '{
+   "version": "1.3",
+   "ciphersSuites": ["TLS_AES_128_GCM_SHA256", "TLS_CHACHA20_POLY1305_SHA256"],
+   "message": "Client Hello"
+}' http://16.170.234.56:8080/clienthello)
+
+
+# Step 2 - Server Hello (Server -> Client)
+SESSION_ID=$(jq -r '.sessionID' <<< "$RESPONSE")
+
+echo "$RESPONSE" | jq -r '.serverCert' > cert.pem
+
+
+# Step 3 - Server Certificate Verification
+wget https://devops-feb23.s3.eu-north-1.amazonaws.com/cert-ca-aws.pem -O cert-ca-aws.pem
+
+VERIFICATION=$(openssl verify -CAfile cert-ca-aws.pem cert.pem)
+
+if [ "$VERIFICATION" != "cert.pem: OK" ]; then
+  echo "Server Certificate is Invalid"
```

Comment:
the correct error should by `Server Certificate is invalid.` (note the capital I).

Reply:
Thanks. It worked 😅

Reply:
Well done, you are the first student that completed the project, amazing job! now go to have some rest...